test/cluster/upsert/02-after-clusterd-restart.td

```diff
@@ -54,20 +54,20 @@ $ kafka-ingest format=avro topic=dbzupsert key-format=avro key-schema=${keyschem
 
 # Verify that the errors from before are still there
 ! SELECT * FROM upsert
-contains: Key decode
-
-# Retract the bad key
-$ kafka-ingest format=bytes topic=dbzupsert key-format=bytes omit-value=true
-broken-key:
-
-# There is still an error, due to the bad value.
-! SELECT * FROM upsert
 contains: Value error
 
 # Update the bad value
 $ kafka-ingest format=avro topic=dbzupsert key-format=avro key-schema=${keyschema} schema=${schema}
 {"id": 2} {"before": null, "after": {"row": {"id": 2, "creature": "cow"}}}
 
+# There is still an error, due to the bad key.
+! SELECT * FROM upsert
+contains: Key decode
+
+# Retract the bad key
+$ kafka-ingest format=bytes topic=dbzupsert key-format=bytes omit-value=true
+broken-key:
+
```

Comment:
Switching from a `HashMap` to a `BTreeMap` for `mz_compute_client::service::PartitionedComputeState::peek_responses` changed the order in which errors get returned in this test. The value error is now returned first, so I had to swap the checks.